docs/self-hosting/self-hosting.md

```diff
@@ -0,0 +1,74 @@
+---
+sidebar_position: 1
+title: Self-Hosting DBOS Apps
+---
+
+Because DBOS is implemented in lightweight TypeScript and Python libraries, you can **run your DBOS application anywhere** as long as it has a Postgres server to connect to.
+DBOS runs entirely inside your process, executing your workflows and recording their state in Postgres.
+By default, when your process restarts, DBOS recovers all its pending workflows.
+However, care must be taken when upgrading your application's code or when managing many DBOS processes at scale to ensure all your workflows are efficiently recovered.
+
+This guide will describe best practices for self-hosting an application that uses DBOS, explaining how to manage workflow recovery both when operating on a single server and when operating at scale.
+
+## Self-Hosting On A Single Server
+
+Self-hosting a DBOS application on a single server is simple: each time you restart your application's process, it recovers all workflows that were executing before the restart (all `PENDING` workflows).
+
+However, it is important to be careful when upgrading your application's code.
+When DBOS is launched, it computes an "application version" from a checksum of the code in your application's workflows (you can override this version through the `DBOS__APPVERSION` environment variable).
+Each workflow is tagged with the version of the application that started it.
+When a DBOS application starts, it does not recover workflows tagged with a different application version.
+Thus, to safely recover workflows started on an older version of your code, you should start a process running that code version.
+
+## Self-Hosting on Multiple Servers
+
+When self-hosting in a distributed setting, it is important to manage workflow recovery so that when an executor crashes, restarts, or is shut down, its workflows are recovered.
+You should assign each executor running a DBOS application an executor ID by setting the `DBOS__VMID` environment variable.
+Each workflow is tagged with the ID of the executor that started it.
+When an application with an executor ID restarts, it only recovers pending workflows assigned to that executor ID.
+You can also instruct your executor to recover workflows assigned to other executor IDs through the [workflow recovery endpoint of the admin API](#workflow-recovery).
+
+It is also important to be careful when upgrading your application's code.
```

Comment:
This is a duplicate. I think we should explain the app version concept in a separate subsection before single server and multiple servers.